src/main/java/codeu/controller/ProfilePageServlet.java

```diff
@@ -1,20 +1,66 @@
 package codeu.controller;
 
+import codeu.model.store.basic.UserStore;
+import codeu.model.store.basic.UserProfileStore;
 import java.io.IOException;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /**
- * Servlet class responsible for User profile pages.
+ * Servlet class responsible for User Profile pages.
  */
 public class ProfilePageServlet extends HttpServlet {
 
+  /** Store class that gives access to Users and User Profiles. */
+  private UserStore userStore;
+  private UserProfileStore userProfileStore;
+
+  /**
+   * Set up state for handling profilepage-related requests. This method is only called when running in a
+   * server, not when running in a test.
+   */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+    setUserProfileStore(UserProfileStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  /**
+   * Sets the UserProfileStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserProfileStore(UserProfileStore userProfileStore) {
+    this.userProfileStore = userProfileStore;
+  }
+
+  /**
+   * This function fires when a user requests the /profilepage URL. It simply forwards the request to
+   * profilepage.jsp.
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) 
     throws IOException, ServletException { 
     request.getRequestDispatcher("/WEB-INF/view/profilepage.jsp").forward(request, response);
   }
 
+  /**
+   * This function fires when a user posts the /profilepage URL. It simply forwards the request to
+   * profilepage.jsp.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
```

Comment:
This is where we can update the user's profile info, right? I'm really not sure, but do you think we include multiple of these for each submit button? (one for profile pic, one for about, one for interests?) I know in the Register Servlet this is where we created the username and password for the user. I can definitely implement this and make a PR but is this where this would happen?

Reply:
I think we could probably do that? We could also handle it in a single post like a "save changes" button, and then it just checks all the fields and updates accordingly

Reply:
It'd probably less confusing to do that I guess? We could always split it up into multiple doPosts if we want to

Reply:
One caveat with multiple doPosts is that the framework only supports one POST handler per page. When you submit the form, your browser sends a POST to the page that was loaded, and the web.xml configuration tells the server that that means it should call this doPost method.

Working around that can be fairly simple: you include a hidden identification value in the form and then use that to delegate from this method to multiple single-use save changes methods. But I recommend you get the single "save all changes" version working first. :)